Polyhedron/demo/Polyhedron/SMesh_type.h

```diff
@@ -0,0 +1,21 @@
+#ifndef SMESH_TYPE_H
+#define SMESH_TYPE_H
+
+#include <boost/scoped_ptr.hpp>
+#include <boost/array.hpp>
+#include <set>
+#include <CGAL/Surface_mesh/Surface_mesh_fwd.h>
```

Comment:
If the header `SMesh_type.h` is suppose to define the type `SMesh`, then the real header should be included!